packages/addons-v5/commands/addons/destroy.js

```diff
@@ -1,40 +1,53 @@
+/* eslint-disable no-await-in-loop */
+
 'use strict'
 
 const cli = require('heroku-cli-util')
+const {notify} = require('../../lib/notify')
 
 async function run(context, heroku) {
+  const destroyAddon = require('../../lib/destroy_addon')
   const resolve = require('../../lib/resolve')
   const {groupBy, toPairs} = require('lodash')
+  const force = context.flags.force || process.env.HEROKU_FORCE === '1'
+  const wait = context.flags.wait || false
 
-  let force = context.flags.force || process.env.HEROKU_FORCE === '1'
   if (context.args.length === 0) throw new Error('Missing add-on name')
 
   let addons = await Promise.all(
     context.args.map(name => resolve.addon(heroku, context.app, name)),
   )
-  for (let addon of addons) {
-    // prevent deletion of app when context.app is set but the addon is attached to a different app
-    let app = addon.app.name
+
+  for (const addon of addons) {
+    // prevent deletion of add-on when context.app is set but the addon is attached to a different app
+    const app = addon.app.name
     if (context.app && app !== context.app) throw new Error(`${cli.color.addon(addon.name)} is on ${cli.color.app(app)} not ${cli.color.app(context.app)}`)
   }
 
   for (let app of toPairs(groupBy(addons, 'app.name'))) {
     addons = app[1]
     app = app[0]
+
     await cli.confirmApp(app, context.flags.confirm)
-    for (let addon of addons) {
-      let msg = `Destroying ${cli.color.addon(addon.name)} on ${cli.color.app(addon.app.name)}`
-      await cli.action(msg, heroku.request({
-        method: 'DELETE',
-        path: `/apps/${addon.app.id}/addons/${addon.id}`,
-        headers: {'Accept-Expansion': 'plan'},
-        body: {force},
-      }))
+
+    for (const addon of addons) {
+      try {
+        await destroyAddon(heroku, addon, force, wait)
+        if (wait) {
+          notify(`heroku addons:destroy ${addon.name}`, 'Add-on successfully deprovisioned')
```

Comment:
What does the user see while waiting here? It looks handled in destroy_addon.js, but I have to ask.

Reply:
When you use the `--wait` flag the command just "hangs up" until the procedure completes, it's the same for any command using that option. This only adds a desktop notification when the command completed, after (possibly) many seconds or minutes.